src/App.js

```diff
@@ -40,6 +45,17 @@ export default function App() {
   }
 
   function handleSubmit(newMedication) {
-    setMedications([newMedication, ...medications])
+    const index = medications.findIndex(
+      medication => medication.id === newMedication.id
+    )
+    if (index > -1) {
+      setMedications([
+        ...medications.slice(0, index),
+        { ...newMedication },
+        ...medications.slice(index + 1),
+      ])
```

Comment:
What do you want to achieve here? I think you want to update a medication if the id exists already? A named function could describe the context here
```suggestion
      updateMedication(index)
```

Put this function below:
```
function updateMedication(index) {
 setMedications([
        ...medications.slice(0, index),
        { ...newMedication },
        ...medications.slice(index + 1),
      ])
}
```

Reply:
good idea! just need to also pass in `newMedication`